Improve numerical stability of GroupNorm backward

xiaomengy opened the pull request:
Summary: Improve numerical stability of GroupNorm backward

Test Plan: buck test mode/opt //caffe2/test:nn -- "GroupNorm_numeric"

Differential Revision: D29654135

vadimkantorov commented on aten/src/ATen/native/cpu/group_norm_kernel.cpp:
```diff
@@ -107,38 +108,19 @@ void ComputeInternalGradients(
-      ds[i] = ds_val;
-      db[i] = db_val;
+      const T mean_val = mean[i / (C / G)];
+      std::tie(ds[i], db[i]) = utils::RowwiseMomentsInternalGraidents<T>(
```
typo: Graidents

vadimkantorov commented on aten/src/ATen/native/cpu/moments_utils.h:
```diff
@@ -143,6 +143,112 @@ std::pair<T, T> RowwiseMoments(const T* X, int64_t N, int64_t ddof = 0) {
 }
 
+template <typename T, int64_t kMaxDepth>
+std::pair<T, T> RowwiseMomentsInternalGraidentsImpl(
```
same as above typo: Graidents

xiaomengy committed:
Improve numerical stability of GroupNorm backward (#61521)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/61521

Improve numerical stability of GroupNorm backward

Test Plan: buck test mode/opt //caffe2/test:nn -- "GroupNorm_numeric"

Differential Revision: D29654135

fbshipit-source-id: 4637e0909b5ad65a087950e458a7ef02adbc88b3
